.ci/Jenkinsfile

```diff
@@ -78,8 +78,8 @@ pipeline {
             steps {
               withGithubNotify(context: "Test-${PLATFORM}") {
                 dir("${BASE_DIR}"){
-                  withGoEnv(){
-                    goTestJUnit(options: '-v ./...', output: 'junit-report.xml')
+                  withMageEnv(){
```

Comment:
If you don't specify `version: "${env.GO_VERSION}"` in the arguments here this will ignore the `.go-version` file and fall back to a default version.

Reply:
I think that's not the case, `.go-version` is used in case `version` is not set, if it cannot be found then it uses a default version, https://github.com/elastic/apm-pipeline-library/blob/main/vars/goDefaultVersion.groovy#L28

Reply:
Hmm, prior to this PR (https://github.com/elastic/elastic-agent-shipper/pull/45/files) we were ignoring the `.go-version` file and defaulting to Go 1.18.2. The console output demonstrating this has been deleted now though.

It seems to work correctly in this PR for now though, maybe I can try to reproduce it separately.

Reply:
Interesting, last change regarding the `.go-version` was done in https://github.com/elastic/apm-pipeline-library/commit/6d8372bbc1c9265b81038c6bb29021f4ed62da08 back in Oct 2021.  I don't know what's the reason https://github.com/elastic/elastic-agent-shipper/pull/45/files was required, but the go-version relies on the below order:

- if `version` argument, then it gets higher precedences
- Otherwise
  - read `.go-version` in the existing folder
  - read `.go-version` in the `BASE_DIR` location
  - use the fixed version defined in https://github.com/elastic/apm-pipeline-library/blob/053dd238f685bf0ed8fc7b576ca0630f5d8bb966/vars/goDefaultVersion.groovy#L41